internal/spec/build.go

```diff
@@ -138,23 +139,17 @@ func FromScript(ctx context.Context, now time.Time, script string) (*flux.Spec,
 	}
 	s.Finish()
 
+	deps := execdeps.NewExecutionDependencies(nil, &now, nil)
+	ctx = deps.Inject(ctx)
+
 	s, cctx := opentracing.StartSpanFromContext(ctx, "eval")
-	sideEffects, scope, err := flux.EvalAST(cctx, astPkg, flux.SetNowOption(now))
+	sideEffects, _, err := flux.EvalAST(cctx, astPkg)
```

Comment:
To clarify, this change (removing `flux.SetNowOption(now)`) introduced a regression. Before this change, the `now` time that was passed in would be used to update the variable environment. Specifically the option `now` would be assigned a new function value.

After this change, the environment is no longer mutated. Instead the `now` time is injected into the execution context. If the `now` option is assigned as part of a flux program, everything works as before. However if a `now` time is passed via the API, because the `now` option is no longer updated in the environment with that new value, the default implementation of `system.time()` will be used.

At the time of this change `system.time()` just ignored the execution context that it was given. Therefore it would always just return the current system time when it was called. To fix this regression, you either still need to pass in `flux.SetNowOption(now)` to `flux.EvalAST`, or you need `system.time()` to check its context for an updated `now` time.

Reply:
@jlapacik I see now, `SetNowOption` was used not just for computing `now` after script evaluation but also in the implementation of `system.time` and I needed to update that as well.

I actually had searched the code for places where Now was pulled from the environment and evaluated, but I didn't include the standard lib in my search.

Reply:
@jlapacik So the call to `SetNowOption` was overriding the

`option now = system.time` 

in universe.flux?

Reply:
Yes.